### PR TITLE
Link fixin

### PR DESCRIPTION
**Pull request by poshul**
Fixed a bunch of stale links, added files for future nase tutorial

### CODE REVIEW

**Comment by poshul** [comment]
Should I be worried about the failing build_test?

**Comment by jpfeuffer** [comment]
It seems we did not have a working pyopenms build in 30? days, so there is not artifact anymore.

**Comment by timosachsenberg** [comment]
there are autofixes available: https://github.com/OpenMS/pyopenms-extra/pull/252/checks?check_run_id=4879571790
does anyone know how trustworthy those are?

**Comment by timosachsenberg** [comment]
If the links work like this I think we can move forward here

**Comment by poshul** [comment]
Lint fails with ` ImportError: cannot import name 'soft_unicode' from 'markupsafe' (/usr/local/lib/python3.7/site-packages/markupsafe/__init__.py)`

**Comment by jpfeuffer** [comment]
This is an error in the action. See their GitHub repo. Should be a comment from me as well. There is a workaround using a fork from someone. But maybe the action was updated by now. Please check.

**Comment by jpfeuffer** [comment]
@timosachsenberg when fixing the API docs I saw quite some warnings like XYZ.rst never referenced anywhere. Including hyper score. Can you check this? Just build yourself locally and you will see.

**Comment by jpfeuffer** [comment]
Unfortunately we have a lot of warning now from the API docs. We should check if we can filter them somehow.
Or do CI without building the API docs, which also takes a few minutes.

**Comment by timosachsenberg** [comment]
Why are these not referenced? What is missing?

File: docs/source/wrap_classes.rst [hunk 1]
```diff
@@ -58,13 +58,13 @@ Open it and add your new function *with the correct indentation*:
   - Warning: For a single-line comment, there should not be a space between wrap-doc and the following comment.
   - Note: The space between the hash and wrap-doc (# wrap-doc) is not necessary, but used for consistency.
   - Note: Please start the comment with a caplital letter.
-  
-See the next section for a SimpleExample_ and a more AdvancedExample_ of a wrapped class with several functions.
+
+See the next section for a `simple example <#Simple-Example>`_ and a more AdvancedExample_ of a wrapped class with several functions.
```
Comment (jpfeuffer) [inline docs/source/wrap_classes.rst:62]: can we have the same style for both links?

Comment (poshul) [reply docs/source/wrap_classes.rst]: yup. fixed

**Comment by timosachsenberg** [comment]
can you revive this so we can move it over the finish line?

**Comment by timosachsenberg** [comment]
cool! thanks

**Pull request merged**